contracts/cryptography/ECDSA.sol

```diff
@@ -33,9 +33,13 @@ library ECDSA {
             v := byte(0, mload(add(signature, 0x60)))
         }
 
-        // Version of signature should be 27 or 28, but 0 and 1 are also possible versions
-        if (v < 27) {
-            v += 27;
+        //EIP-2 still allows signature malleabality for ecrecover(), remove this possibility and make
+        //the signature unique. Most signatures from current libraries are generate a unique signature with
+        //an s-value in the lower half order. If not, calculate a new s-value with
+        //0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141 - s1 and flip v from 27 to 28 or
+        //vice versa. It could also happen that a signature uses 0/1 instead 27/28 for v. In that case add 27.
```

Comment:
```suggestion
        //vice versa.
```

Reply:
@nventuro No worries, good things take time.

Here I wanted to write what you need to do if you have a signature, which is now not valid anymore due to accepting only unique signatures. On the one hand, you can flip s, but there could also be signatures with 0/1 for v, in which case you need to add 27. I                                                                        was also thinking about having to functions:

function recover(...) -> only accepts unique signatures

and

function recoverMalleable(...) -> the one that accepts both s and v.

Reply:
Ah, I read that comment as 'in that case, we add 27', which is what the old code did, as opposed to a suggestion to a reader. We may want to be more explicit here, e.g. 'if you have such a signature, it can be made valid by adding 27 to `v`'.